packages/lime-mesh-upgrade/files/usr/lib/lua/lime-mesh-upgrade.lua

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env lua
+
+local eupgrade = require 'eupgrade'
+local config = require "lime.config"
+local utils = require "lime.utils"
+local network = require("lime.network")
+
+
+local mesh_upgrade = {
+    -- posible tranactin states
+    transaction_states = {
+        NO_TRANSACTION = "no_transaction",
+        STARTED = "started", -- there is a transaction in progress
+        ABORTED = "aborted",
+        FINISHED = "finished"
+    },
+    -- psible upgrade states enumeration
+    upgrade_states = {
+        DEFAULT = "not upgrading",
+        STARTING = "starting",
+        DOWNLOADING = "downloading",
+        READY_FOR_UPGRADE = "ready_for_upgrade",
+        UPGRADE_SCHELUDED = "upgrade_scheluded",
+        CONFIRMATION_PENDING = "confirmation_pending",
+        CONFIRMED = "confirmed",
+        UPDATED = "updated",
+        ERROR = "error"
+    },
+    -- list of possible errors
+    errors = {
+        DOWNLOAD_FAILED = "download failed",
+        CONFIRMATION_TIME_OUT = "confirmation timeout"
+    }
+}
+
+--shoud epgrade be disabled ?
+eupgrade.set_workdir("/tmp/mesh_upgrade")
```

Comment:
We have to implement this function? 

On my local code, before puling changes i havve this code that i was working with:

```ts
function mesh_upgrade.set_workdir(workdir)
    if not utils.file_exists(workdir) then
        os.execute('mkdir -p ' .. workdir)
    end
    if fs.stat(workdir, "type") ~= "dir" then
        error("Can't configure workdir " .. workdir)
    end
    mesh_upgrade.WORKDIR = workdir
    --mesh_upgrade.DOWNLOAD_INFO_CACHE_FILE = mesh_upgrade.WORKDIR .. '/download_status'
    mesh_upgrade.FIRMWARE_LATEST_JSON = mesh_upgrade.WORKDIR .. "/firmware_latest_mesh_wide.json"
    --mesh_upgrade.FIRMWARE_LATEST_JSON_SIGNATURE = mesh_upgrade.FIRMWARE_LATEST_JSON .. '.sig'
end
```

Maybe this code is lost on a push?

Reply:
After look a little bit on the code, and blame the file I found that this commit https://github.com/selankon/lime-packages/commit/d7d3d304726daf719cec1a354ffcdd900d1879c9 comments the above function and this one deletes it https://github.com/selankon/lime-packages/commit/b7dd972deaa32cdd5667ab532eaa9152db4d0478.

But the function `mesh_upgrade.set_up_firmware_repository` needs what this code do: 

https://github.com/selankon/lime-packages/pull/2/files#diff-93ac8eccfc608a16efdfe8be36a18efc8bccfae4fdb52ae2e22ee8f0498e4a09R61

So I supose that is disabled by error, i will add it again on a future commit

Reply:
I thought about using the eupgrade infrastructure directly, if it that is possible, the code mentioned above is not necessary anymore, 
Also, one of the changes incorporates the report of the downloaded firmware local file url.  
I added a new method to eupgrade to change the repository url. 

But this is just a proposal, We can make a call to discuss this, Maybe I am missing something.

Reply:
hoo I see, maybe this file can be the same from eupgrade eupgrade.FIRMWARE_LATEST_JSON in this way we dont have to implement the same code here. Just as a suggestion